test-data/unit/check-isinstance.test

```diff
@@ -602,6 +638,14 @@ while 1:
 else:
     x + 1
 x + 1      # E: Unsupported operand types for + (likely involving Union)
+x = 1
+for y in [1]:
+    x + 1
+    x = 'a'
+    break
+else:
+    x + 1
```

Comment:
Do you have a test like this one (and/or the `while` loop in the same test case) where the `break` in the loop is conditional (say on the result of `foo()`) so the body of the `else` clause is ill-typed?